chirp-impl/src/main/java/sample/chirper/chirp/impl/JpaSessionImpl.java

```diff
@@ -0,0 +1,67 @@
+package sample.chirper.chirp.impl;
+
+import akka.Done;
+import com.lightbend.lagom.internal.persistence.jdbc.SlickProvider;
+import play.inject.ApplicationLifecycle;
+import scala.compat.java8.FutureConverters;
+import scala.compat.java8.JFunction0;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.EntityTransaction;
+import javax.persistence.Persistence;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Function;
+
+@Singleton
+public class JpaSessionImpl implements JpaSession {
+    private final EntityManagerFactory factory;
+    private final SlickProvider slick;
+
+    @Inject
+    JpaSessionImpl(SlickProvider slick, ApplicationLifecycle lifecycle) {
+        this.factory = Persistence.createEntityManagerFactory("default");
```

Comment:
I'm not totally positive if this is the best/safest place to initialize the `EntityManagerFactory` ... it may require a connection to the database, which may mean that the service fails to start if it's not yet available.

I'll need to test this with a non-embedded DB such as PostgreSQL or MySQL to see what happens.

If that is indeed the case, what's the best approach? An actor? Looking deeper, maybe it isn't necessary because this depends on `SlickProvider`, and that depends on `DBApi` with the comment "Ensures database is initialised before we start anything that needs it". Does that mean that this needs to await `slick.ensureTablesCreated()` though?